### DISCUSSION TITLE
Expand usage of `Craft::parseBooleanEnv()` to general.php

### DISCUSSION THREAD

**Discussion by aaronbushnell** [craftcms/cms#10319] [Ideas] [answered]
Right now it doesn't seem like you can use `Craft::parseBooleanEnv()` in `config/general.php` to parse boolean-based environment variables. Here's an example:

## Desired:
```php
<?php

use Craft;

return [
  '*' => [
    'devMode' => Craft::parseBooleanEnv('$DEV_MODE'),
  ],
];
```

Doing this will throw a fatal error (I truncated my folder paths out of here):

### Error
```
Fatal error: Uncaught Error: Class 'Craft' not found in /../config/general.php:7
Stack trace: #0 /../vendor/craftcms/cms/src/services/Config.php(205): include()
#1 /../vendor/craftcms/cms/bootstrap/bootstrap.php(169): craft\services\Config->getConfigFromFile('general')
#2 /../vendor/craftcms/cms/bootstrap/web.php(51): require('/-...')
#3 /../web/index.php(21): require('/-...')
#4 /../.composer/vendor/laravel/valet/server.php(235): require('/-...')
#5 {main} thrown in /../config/general.php on line 7
```

Right now you can get around that, but it requires writing an expression like this

## Current
```php
<?php

use craft\helpers\App;

return [
  '*' => [
    'devMode' => App::env('DEV_MODE') === "true",
  ],
];
```

**Comment by brandonkelly** [comment, marked as answer]
This was a tricky one! `config/general.php` needs to be loaded _before_ `Craft.php` (where the `Craft` class is defined) because we need to know whether Dev Mode should be enabled before loading Craft/Yii, so we can declare the `YII_DEBUG` constant based on it. Otherwise, Yii will declare the `YII_DEBUG` constant for us, setting it to `false`, and the `devMode` config value will have no effect.

So… just worked around this by moving `parseEnv()` and `parseBooleanEnv()` over to `craft\helpers\App` for the next release, where they can be called before Craft/Yii is loaded:

```php
use craft\helpers\App;

return [
  '*' => [
    'devMode' => App::parseBooleanEnv('$DEV_MODE'),
  ],
];
```

**Reply by aaronbushnell** [reply]
Thanks so much, @brandonkelly! I had a feeling it had something to do with the load order (or maybe I was misunderstanding the use of the function). Appreciate it a bunch! 🙌

**Reply by brandonkelly** [reply]
Craft 3.7.29 is out now with this change!
